### PR TITLE
Bug 2084079: Refactors CreateRouteIfNotExists to CreateOrUpdateRoute

### PR DESCRIPTION
**Pull request by JoaoBraveCoding**
Issue: https://bugzilla.redhat.com/show_bug.cgi?id=2084079

Problem: on OpenShift 4.11 the Prometheus route was updated with
a subpath /api, CMO was only able to create routes and not update them.
When users would update from a version before 4.11 to 4.11 the route
would not be updated.

Solution: allow CMO to Create or Update Routes similar to how it
interects with other k8s resources.

* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: pkg/client/client.go [hunk 1]
```diff
@@ -346,14 +346,27 @@ func (c *Client) CreateOrUpdateSecurityContextConstraints(ctx context.Context, s
...
+	existing, err := rclient.Get(ctx, r.GetName(), metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
 		_, err := rclient.Create(ctx, r, metav1.CreateOptions{})
 		return errors.Wrap(err, "creating Route object failed")
 	}
-	return nil
+	if err != nil {
+		return errors.Wrap(err, "retrieving Route object failed")
+	}
+	if reflect.DeepEqual(r.Spec, existing.Spec) {
```
Comment (simonpasquier) [inline pkg/client/client.go:359]: I think that the `spec.host` will always be different because it's automatically chosen if unset.

**Review by simonpasquier — COMMENTED** [review]
Great! minor comments only :)

File: pkg/client/client.go (hunk 1 above)
Comment (jan--f) [reply pkg/client/client.go:359]: @simonpasquier is [correct](https://github.com/openshift/api/blob/a99ffa1cac6709edf8f502b16890b16f9a557e00/route/v1/types.go#L75-L80).

Dropping this code path probably makes most sense?

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Pull request merged**